### PR TITLE
[FIX]  Refactory Calendar Event module Test case code

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

File: project_resource_calendar/tests/test_calendar_event.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class TestCalendarEvent(TestCalendarEventCommon):
+
+    def setUp(self):
+        super(TestCalendarEvent, self).setUp()
+        self.validate_room_name = 'Testing_Room_type_2'
+        self.validate_instrument_name = 'Testing_Instrument_type_2'
+
+    def test_010_onchange_room_id(self):
```
Comment (eilst) [inline project_resource_calendar/tests/test_calendar_event.py:34]: To test on_change method you should validate the state before and after and validate that what you are changing actually changed ( the purpose of the on change room id method). Maybe I don’t get it, but what i’m see here, is that you are only asserting the name after create, not on_change

File: project_resource_calendar/tests/common.py [hunk 2]
```diff
@@ -8,7 +8,12 @@ class TestCalendarEventCommon(common.TransactionCase):
 
     def setUp(self):
         super(TestCalendarEventCommon, self).setUp()
-
+        # Initialization of param values
+        room_name = 'Testing_Room_type_2'
+        resource_type_name = 'room'
+        instrument_name = 'Testing_Instrument_type_2'
+        instrument_type_name = 'room'
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/common.py]: instrument type is not 'room'

File: project_resource_calendar/tests/common.py [hunk 3]
```diff
@@ -8,7 +8,12 @@ class TestCalendarEventCommon(common.TransactionCase):
 
     def setUp(self):
         super(TestCalendarEventCommon, self).setUp()
-
+        # Initialization of param values
+        room_name = 'Testing_Room_type_2'
+        resource_type_name = 'room'
+        instrument_name = 'Testing_Instrument_type_2'
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/common.py]: why the name value is separated by  : '_'  room or instrument name is a string so you can separate words by space.

File: project_resource_calendar/tests/common.py [hunk 4]
```diff
@@ -8,7 +8,12 @@ class TestCalendarEventCommon(common.TransactionCase):
 
     def setUp(self):
         super(TestCalendarEventCommon, self).setUp()
-
+        # Initialization of param values
+        room_name = 'Testing_Room_type_2'
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/common.py]: this variable is used only one time you can directly put the room name in the creation of the resource.calendar.room object

File: project_resource_calendar/tests/common.py [hunk 5]
```diff
@@ -19,16 +24,25 @@ def setUp(self):
...
             'allow_double_book': True,
         })
-    
         self.weekly_report_wizard_common = self.WeeklyReportWizards.create({
                 'room_id': self.room_1.id,
                 'recurrency': True,
                 'state': 'open',
             })
+        self.room_2 = self.Rooms.create({
+            'name': room_name,
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/common.py]: You can directly put the name without using this variable

File: project_resource_calendar/tests/common.py [hunk 6]
```diff
@@ -19,16 +24,25 @@ def setUp(self):
...
+            'resource_type': resource_type_name,
+            'allow_double_book': double_book_status,
+        })
+        self.instrument_2 = self.Instruments.create({
+            'name': instrument_name,
+            'resource_type': instrument_type_name,
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/common.py]: check the resource type it should be a material

File: project_resource_calendar/tests/test_calendar_event.py (hunk 1 above)
Comment (rimbendhaou) [reply project_resource_calendar/tests/test_calendar_event.py:34]: you have to check the onchange_room_id method and to check the values before and after the execution of this method

File: project_resource_calendar/tests/test_instrument.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+                          'model_resource_calendar_instrument'
+
+        self.room_info = self.Rooms.create({
+            'name': 'Test Room 1',
+            'resource_type': 'room',
+            'allow_double_book': True,
+        })
+        self.instrument_type = self.Instruments.create({
+            'name':  self.instrument_name,
+            'resource_type':  self.resource_type_name,
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_instrument.py]: resource type is material

File: project_resource_calendar/tests/test_instrument.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        self.double_book_status = True
+        self.model_name = 'project_resource_calendar.' \
+                          'model_resource_calendar_instrument'
+
+        self.room_info = self.Rooms.create({
+            'name': 'Test Room 1',
+            'resource_type': 'room',
+            'allow_double_book': True,
+        })
+        self.instrument_type = self.Instruments.create({
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_instrument.py]: self.Instruments.create should return an instrument so better to name the variable as an instrument not instrument_type

File: project_resource_calendar/tests/test_instrument.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        self.instrument_name = 'Instrument_name_for_auditlog_testing'
+        self.resource_type_name = 'material'
+        self.double_book_status = True
+        self.model_name = 'project_resource_calendar.' \
+                          'model_resource_calendar_instrument'
+
+        self.room_info = self.Rooms.create({
+            'name': 'Test Room 1',
+            'resource_type': 'room',
+            'allow_double_book': True,
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_instrument.py:18]: Why some time you use the self.double_book_status and some time you directly put True. I prefer that you directly put True or False without using a variable.

File: project_resource_calendar/tests/test_room.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    def setUp(self):
+        super(TestRoom, self).setUp()
+        self.check_record_count = 2
+        self.room_name = 'Room_name_for_auditlog_testing'
+        self.resource_type_name = 'room'
+        self.double_book_status = True
+        self.model_name = 'project_resource_calendar.' \
+                          'model_resource_calendar_room'
+
+        self.room_type = self.Rooms.create({
```
Comment (rimbendhaou) [inline project_resource_calendar/tests/test_room.py]: it is a room not a room_type

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_resource_calendar/tests/test_room.py (hunk 10 above)
Comment (jananjoy) [reply project_resource_calendar/tests/test_room.py]: @rimbendhaou : Done

File: project_resource_calendar/tests/test_instrument.py (hunk 8 above)
Comment (jananjoy) [reply project_resource_calendar/tests/test_instrument.py]: @rimbendhaou : Done

File: project_resource_calendar/tests/test_instrument.py (hunk 9 above)
Comment (jananjoy) [reply project_resource_calendar/tests/test_instrument.py:18]: @rimbendhaou :Done

File: project_resource_calendar/tests/test_instrument.py (hunk 7 above)
Comment (jananjoy) [reply project_resource_calendar/tests/test_instrument.py]: @rimbendhaou : Done

File: project_resource_calendar/tests/common.py (hunk 6 above)
Comment (jananjoy) [reply project_resource_calendar/tests/common.py]: @rimbendhaou Done

File: project_resource_calendar/tests/common.py (hunk 5 above)
Comment (jananjoy) [reply project_resource_calendar/tests/common.py]: @rimbendhaou Done

File: project_resource_calendar/tests/common.py (hunk 4 above)
Comment (jananjoy) [reply project_resource_calendar/tests/common.py]: @rimbendhaou Done

File: project_resource_calendar/tests/common.py (hunk 3 above)
Comment (jananjoy) [reply project_resource_calendar/tests/common.py]: @rimbendhaou : use to remove the wildcard but changed it

File: project_resource_calendar/tests/common.py (hunk 2 above)
Comment (jananjoy) [reply project_resource_calendar/tests/common.py]: @rimbendhaou Done

File: project_resource_calendar/tests/test_calendar_event.py (hunk 1 above)
Comment (jananjoy) [reply project_resource_calendar/tests/test_calendar_event.py:34]: Change it

**Review by rimbendhaou — APPROVED** [review]

**Pull request merged**